conda_pypi/main.py

```diff
@@ -168,3 +194,138 @@ def ensure_target_env_has_externally_managed(command: str):
                 path.unlink()
     else:
         raise ValueError(f"command {command} not recognized.")
+
+
+def pypi_lines_for_explicit_lockfile(
+    prefix: Path | str, checksum: Literal["md5", "sha256"] | None = None
+) -> list[str]:
+    PrefixData._cache_.clear()
+    pd = PrefixData(str(prefix), pip_interop_enabled=True)
+    pd.load()
+    lines = []
+    python_record = list(pd.query("python"))
+    assert len(python_record) == 1
+    python_record = python_record[0]
+    python_details = {"version": ".".join(python_record.version.split(".")[:3])}
+    if "pypy" in python_record.build:
+        python_details["implementation"] = "pp"
+    else:
+        python_details["implementation"] = "cp"
+    for record in pd.iter_records():
+        if record.package_type != PackageType.VIRTUAL_PYTHON_WHEEL:
+            continue
+        ignore = False
+        wheel = {}
+        hashed_record = ""
+        for path in record.files:
+            path = Path(context.target_prefix, path)
+            if "__editable__" in path.stem:
+                ignore = True
+                break
+            if path.name == "direct_url.json" and path.parent.suffix == ".dist-info":
+                data = json.loads(path.read_text())
```

Comment:
this could raise OSError with permissions errors etc and should be handled more defensively